pkg/controllers/workspace_controller.go

```diff
@@ -155,6 +179,81 @@ func (c *WorkspaceReconciler) deleteWorkspace(ctx context.Context, wObj *kaitov1
 	return c.garbageCollectWorkspace(ctx, wObj)
 }
 
+func (c *WorkspaceReconciler) updateControllerRevision(ctx context.Context, wObj *kaitov1alpha1.Workspace) error {
+	currentHash := computeHash(wObj)
+	annotations := wObj.GetAnnotations()
+
+	latestHash, exists := annotations[WorkspaceRevisionAnnotation]
+	if exists && latestHash == currentHash {
+		return nil
+	}
+
+	data := map[string]string{"hash": currentHash}
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return fmt.Errorf("failed to marshal revision data: %w", err)
+	}
+
+	revisions := &appsv1.ControllerRevisionList{}
+	if err := c.List(ctx, revisions, client.InNamespace(wObj.Namespace), client.MatchingLabels{WorkspaceNameLabel: wObj.Name}); err != nil {
+		return fmt.Errorf("failed to list revisions: %w", err)
+	}
+	sort.Slice(revisions.Items, func(i, j int) bool {
+		return revisions.Items[i].Revision > revisions.Items[j].Revision
+	})
```

Comment:
Could also add the latest revision number in the annotations...this would prevent need to sort